rest/interceptors/src/main/java/com/connexta/spring/interceptor/VersionInterceptor.java

```diff
@@ -0,0 +1,276 @@
+ * included in the generated spring API using Apache's ServiceMix depends maven plugin.
+ */
+@Component
+public class VersionInterceptor implements HandlerInterceptor {
+  private static final Logger LOGGER = LoggerFactory.getLogger(VersionInterceptor.class);
+
+  private static final String ACCEPT_VERSION = "Accept-Version";
+  private static final String CONTENT_VERSION = "Content-Version";
+
+  public static final String DEPENDENCIES_FILE = "META-INF/maven/dependencies.properties";
+  public static final String VERSION = "version";
+
+  private static final Map<Method, String> versions = new ConcurrentHashMap<>();
+
+  @Override
+  public boolean preHandle(
+      HttpServletRequest request, HttpServletResponse response, Object handler) {
+    if (handler instanceof HandlerMethod) {
+      final HandlerMethod handlerMethod = (HandlerMethod) handler;
+      final String serverVersion = VersionInterceptor.getVersion(handlerMethod.getMethod());
+      final String clientVersion = request.getHeader(VersionInterceptor.ACCEPT_VERSION);
+
+      LOGGER.debug(
+          "VersionInterceptor.preHandle(handler: {}, server: {}, client: {}",
+          handlerMethod.getShortLogMessage(),
+          serverVersion,
+          clientVersion);
+      if ((serverVersion != null)
+          && !VersionInterceptor.areCompatible(serverVersion, clientVersion)) {
+        throw new UnsupportedVersionException(clientVersion, serverVersion);
+      }
+    }
+    return true;
+  }
+
+  @Override
+  public void postHandle(
+      HttpServletRequest request,
+      HttpServletResponse response,
+      Object handler,
+      ModelAndView modelAndView) {
+    if (handler instanceof HandlerMethod) {
+      final HandlerMethod handlerMethod = (HandlerMethod) handler;
+      final String serverVersion = VersionInterceptor.getVersion(handlerMethod.getMethod());
+
+      LOGGER.debug(
+          "VersionInterceptor.postHandle(handler: {}, server: {}",
+          handlerMethod.getShortLogMessage(),
+          serverVersion);
+      if (serverVersion != null) {
+        response.addHeader(VersionInterceptor.CONTENT_VERSION, serverVersion);
+      }
+    }
+  }
+
+  private static boolean areCompatible(String serverVersion, String clientVersion) {
+    if (serverVersion.equals(clientVersion)) {
+      return true;
+    }
+    final String[] serverParts = serverVersion.split("\\.");
+    final String[] clientParts = clientVersion.split("\\.");
+    boolean result = true;
+
+    for (int i = 0; result && (i < 2); i++) {
+      final String serverPart = (serverParts.length > i) ? serverParts[i].trim() : "0";
+      final String clientPart = (clientParts.length > i) ? clientParts[i].trim() : "0";
+
+      result = serverPart.equals(clientPart);
+      if (result) {
+        continue;
+      }
+      final Integer serverValue = Integer.parseInt(serverPart); // let parsing errors bubble out
```

Comment:
Like any other errors it would be an internal server error unless you provide exception handlers to change it to something else. In the latest version, I am purposely not catching the server version exception as this would be an internal server error and I am catching the client one so I can handle it more appropriately.